csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -871,6 +980,15 @@ public new void Abort(Ice.Exception ex)
             base.Abort(ex);
         }
 
+        public new void Abort(Ice.RemoteException ex)
+        {
+            Ice.InvocationMode mode = Proxy.IceReference.GetMode();
+
+            Debug.Assert(mode != Ice.InvocationMode.BatchOneway &&
+                         mode != Ice.InvocationMode.BatchDatagram); // not implemented
+            base.Abort(ex);
+        }
```

Comment:
This isn't necessary, Abort is only called from the user thread when the client makes an invocation on the proxy.

Reply:
Fixing.